firmware/quadruna/VC/src/app/vehicle_dynamics/app_pid.c

```diff
@@ -22,8 +25,20 @@ void app_pid_init(PID *pid, const PID_Config *conf)
 float app_pid_compute(PID *pid, float setpoint, float input)
 {
     pid->error = setpoint - input;
-    pid->integral += pid->error;
+
+    if(app_guard_init(pid)){ // if initialized implement the gaurd
+        if(pid->integral_windup_max > pid->integral && pid->integral > pid->integral_windup_min){
+            pid->integral += pid->error;
+        }
+    } 
+    else{
+        pid->integral += pid->error;
+    }
+    
+    // removing filtering for now 
+
     pid->derivative = (input - pid->prev_input);
+
     pid->prev_input = input;
     float output    = pid->Kp * pid->error + pid->Ki * pid->integral - pid->Kd * pid->derivative;
```

Comment:
Why is this minus derivative btw

Reply:
I thought you and Kyle had a reason for doing that

Reply:
This was before my time 😀

Reply:
You may need to fix pid cuz not sure what was written before but I don't think this makes sense